recipes/melon/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+        self.requires("fmt/[>=10.0.0]", transitive_headers=True)
+
+    # same package ID for any package
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            # Validate the minimum cpp standard supported when installing the package. For C++ projects only
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        # In case this library does not work in some another configuration, it should be validated here too
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f"{self.ref} can not be used on Windows.")
+
+    def source(self):
+        # Download source package and extract to source folder
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    # Not mandatory when there is no patch, but will suppress warning message about missing build() method
+    def build(self):
+        # The attribute no_copy_source should not be used when applying patches in build
+        apply_conandata_patches(self)
+
+    # Copy all files to the package folder
+    def package(self):
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        copy(
+            self,
+            "*.hpp",
+            os.path.join(self.source_folder, "include"),
+            os.path.join(self.package_folder, "include"),
+        )
+
+    def package_info(self):
+        # Folders not used for header-only
+        self.cpp_info.bindirs = []
+        self.cpp_info.libdirs = []
+
+        # Set these to the appropriate values if the package has an official FindPACKAGE.cmake
+        # listed in https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules
+        # examples: bzip2, freetype, gdal, icu, libcurl, libjpeg, libpng, libtiff, openssl, sqlite3, zlib...
+        # self.cpp_info.set_property("cmake_module_file_name", "melon")
+        # self.cpp_info.set_property("cmake_module_target_name", "melon::melon")
+
+        # Set these to the appropriate values if package provides a CMake config file
+        # (package-config.cmake or packageConfig.cmake, with package::package target, usually installed in <prefix>/lib/cmake/<package>/)
+        # self.cpp_info.set_property("cmake_file_name", "melon")
+        # self.cpp_info.set_property("cmake_target_name", "melon::melon")
+        # Set this to the appropriate value if the package provides a pkgconfig file
+        # (package.pc, usually installed in <prefix>/lib/pkgconfig/)
+        # self.cpp_info.set_property("pkg_config_name", "melon")
```

Comment:
Feel free to remove these lines, alongside the rest of the comments from the template :)